iotfunctions/anomaly.py

```diff
@@ -204,19 +152,53 @@ def execute(self, df):
                 freqsTS = freqsTS * freqsTSb
                 freqsTS[freqsTS == 0] = 1 / self.windowsize
 
-                # Compute energy = frequency * spectral density over time in decibel - no log10
-                ETS = np.dot(SxTS.T, freqsTS)
+                highfreqsTS = freqsTS.copy()
+                lowfreqsTS = freqsTS.copy()
+                highfreqsTS[highfreqsTS <= 0.3] = 0
+                lowfreqsTS[lowfreqsTS > 0.3] = 0
 
-                # compute zscore over the energy
-                ets_zscore = (ETS - ETS.mean())/ETS.std(ddof=0)
-                logger.debug('NoData z-score max: ' + str(ets_zscore.max()))
+                # Compute energy = frequency * spectral density over time in decibel
+                try:
+                    lowETS = np.log10(np.dot(SxTS.T, lowfreqsTS))
+                    highETS = np.log10(np.dot(SxTS.T, highfreqsTS))
+                    ETS = np.log10(np.dot(SxTS.T, freqsTS))
```

Comment:
what is ETS ? is it constant?

Reply:
renamed to signal_energy  -  from E(nergy)T(emperature)S(ignal)